Sources/SwiftDocC/Model/DocumentationNode.swift

```diff
@@ -849,3 +852,45 @@ private extension BlockDirective {
         directivesSupportedInDocumentationComments.contains(name)
     }
 }
+
+extension [String] {
+
+    /// Strip the minimum leading whitespace from all the strings in this array, as follows:
+    /// - Find the line with least amount of leading whitespace. Ignore blank lines during this search.
+    /// - Remove that number of whitespace chars from all the lines (including blank lines).
+    /// - Returns: An array of strings with the minimum leading whitespace removed.
+    func linesWithoutLeadingWhitespace() -> [String] {
+
+        // Optimization: If the array of lines is empty then we return
+        guard !isEmpty else {
+            return self
+        }
+
+        // Optimization for the common case: If any of the lines do not start
+        // with whitespace, return the original lines.
+        for line in self {
+            if let firstChar = line.first, !firstChar.isWhitespace {
+                return self
+            }
+        }
+
+        /// - Count the leading whitespace characters in the given string.
+        /// - Returns: The count of leading whitespace characters, if the string is not blank,
+        ///     or `nil` if the string is empty or blank (contains only whitespace)
+        func leadingWhitespaceCount(_ line: String) -> Int? {
+            let count = line.prefix(while: \.isWhitespace).count
+            guard count < line.count else { return nil }
+            return count
+        }
+
+        // Find the minimum count of leading whitespace. If there are no
+        // leading whitespace counts (if all the lines were blank) then return
+        // the original lines.
+        guard let minimumWhitespaceCount = self.compactMap(leadingWhitespaceCount).min() else {
+            return self
+        }
```

Comment:
minor: I don't know if the optimizer will remove the temporary array or not but if you want to be sure you can avoid creating a temporary array and instead computing the minimum using a for-loop or a call to `reduce(_:_:)`

```suggestion
        let minimumWhitespaceCount = self.reduce(Int?.none) { partialResult, line in
            guard let leadingWhitespace = leadingWhitespaceCount(line) else {
                return partialResult
            }
            return Swift.min(partialResult ?? .max, leadingWhitespace)
        }! // If there were no lines or if all lines were empty this function would have done an early exit above.
```

Reply:
Again, since the benchmarks look good I think it's better to keep the original which is a lot more readable in my opinion. But nice work to avoid the extra array allocation. It's clever how `reduce` repeatedly mutates the return value array without any allocations after the first one.

If we were routinely processing large arrays of 1000s of lines this sort of optimization might be worth it.

Reply:
Yes, like I said those were only suggestions of things to look at if this code turned out to have a a measurable performance impact.